gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/handler/context/ContextAttributeMap.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.jupiter.reactor.handler.context;
+
+import static io.gravitee.gateway.jupiter.api.context.ExecutionContext.ATTR_PREFIX;
+
+import java.util.HashMap;
+
+/**
+ * Special {@link java.util.Map} implementation allowing to retrieve attributes prefixed with #ATTR_PREFIX without having to specify it explicitly.
+ */
+class ContextAttributeMap extends HashMap<String, Object> {
```

Comment:
should it be good to have it in `gateway-api` ?

Reply:
I don't think as it is an implementation. Also I would prefer to keep it package protected.